### ISSUE TITLE
Rename `SynchronizedIterator` to `SynchronizedAdvancer`

### ISSUE THREAD

**Issue by Adda0** [VeriFIT/mata#314]
Because `SynchronizedIterator` is not an iterator in a C++-sense, we should rename it to something else. We thought about `SynchronizedAdvancer`, seeing that it already has the `advance()` method.

**Comment by tfiedor** [comment]
Can't this simply be e.g. `SynchronizedTraversal`? I mean that is at least valid and existing word.

**Comment by Adda0** [comment]
*advancer* is a valid word, too. See [dictionary records for advancer](https://www.wordnik.com/words/advancer), for example ;) And I must say that I like advancer. It makes it clearer to me what the class does than anything else, to be honest. What do the others think?

**Comment by tfiedor** [comment]
https://www.oxfordlearnersdictionaries.com/spellcheck/english/?q=advancer
https://dictionary.cambridge.org/dictionary/english/advancer
I rest my case.

**Comment by Adda0** [comment]
Nah, who would believe Oxford or Cambridge dictionaries these days, anyway? I believe in *GNU version of the Collaborative International Dictionary of English* :joy:
